glprogs/ambientInteraction.fs

```diff
@@ -90,6 +90,7 @@ void main() {
 		light.rgb *= matDiffuse.rgb * light1;
 	}
 
+	light = max(light, vec4(0));  // avoid negative values, which with floating point render buffers can lead to NaN artefacts
```

Comment:
One would think so, and yet it happened. I didn't actually observe truly negative numbers, but in some rare cases there were rounding errors when nSight would then show the texture content as (-0.000...) instead of just (0), and that was enough to produce NaN and visual artifacts in coming calculations